plugins/OnlineSales.Plugin.SendGrid/Controllers/SendgridController.cs

```diff
@@ -0,0 +1,281 @@
+﻿// <copyright file="SendgridController.cs" company="WavePoint Co. Ltd.">
+// Licensed under the MIT license. See LICENSE file in the samples root for full license information.
+// </copyright>
+
+using System.Collections.Immutable;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using OnlineSales.Data;
+using OnlineSales.Entities;
+using OnlineSales.Interfaces;
+using OnlineSales.Plugin.SendGrid.Data;
+using OnlineSales.Plugin.SendGrid.DTOs;
+using OnlineSales.Plugin.SendGrid.Entities;
+using OnlineSales.Plugin.SendGrid.Exceptions;
+using SendGrid.Helpers.EventWebhook;
+using Serilog;
+
+namespace OnlineSales.Plugin.SendGrid.Controllers;
+
+[Route("api/[controller]")]
+public class SendgridController : ControllerBase
+{
+    private static readonly int BatchSize = 10;
+
+    private static readonly DateTime UnixTimestampZeroDate = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
```

Comment:
@IgorKostyukov this constant is already used in two places. I think it is time to move it somewhere to the common place. Probably together with all the helper methods dependent on that constant.